server/src/main/java/org/elasticsearch/search/DocValueFormat.java

```diff
@@ -208,6 +210,20 @@ public DateTime(StreamInput in) throws IOException {
                 this.timeZone = ZoneId.of(zoneId);
                 this.resolution = DateFieldMapper.Resolution.ofOrdinal(in.readVInt());
             }
+            if (in.getVersion().onOrAfter(Version.CURRENT)) {
+                //if stream is from 7.7 it will have a flag indicating if format is joda
+                boolean isJoda = in.readBoolean();
```

Comment:
This could be refactored to set isJoda based on version/joda pattern check and then always use the conditional on the next line.

Reply:
not sure.. you suggest not sending this boolean flag?
what about example  ike:
DocValuFormat using JodaFormatter - it was able to tell that by checking index creation version. It is serialized (writeTo) in 7.7 
Then deserialisation happens in 7.7 I don't have information about index creation. Pattern itself is not enough as the same pattern can be used in java and joda.

Reply:
The boolean is fine. My suggestion is to reorganize it a bit to remove duplication:

```
final boolean isJoda;
if (in.getVersion().onOrAfter(Version.V_7_7_0)) {
    isJoda = in.readBoolean();
} else {
    isJoda = Joda.isJodaPattern(in.getVersion(), datePattern);
}
this.formatter = isJoda ? Joda.forPattern(datePattern) : DateFormatter.forPattern(datePattern);

```

Reply:
good idea